### ISSUE TITLE
Use floating tabbar on show-on-switching mode

### ISSUE THREAD

**Issue by Dietr1ch** [The-Compiler/qutebrowser#1593]
This should ease redraws and also remove the temporal quick movement on tab switching.

**Comment by The-Compiler** [comment]
I'm not sure I understand the issue - do you mean the webpage should not resize when the tabbar is shown/hidden?

**Comment by Dietr1ch** [comment]
Yes. The resize moves the top of the page down making it move and go back to the full window size for a fraction of a second. Vertical tabbar causes a larger resize.
It should work as the "ex-command" menu, over the page instead of shrinking the page area, causing no page reflows.

**Comment by davidsierradz** [comment]
@The-Compiler Hi, any plans on this? I really dont like how the page area resize all the times the tab bar is switched, thanks for a really good browser.

**Comment by The-Compiler** [comment]
This isn't a priority for me right now, sorry. Contributions welcome, though!

**Comment by The-Compiler** [comment]
Same request for the status bar: #6647

**Comment by The-Compiler** [comment]
Similarly, for the download bar: https://www.reddit.com/r/qutebrowser/comments/ym6fn2/is_it_possible_to_autohide_the_tab_bar/iv2bwot/

**Comment by I03D** [comment]
When the status bar is at the top, it's even more annoying. Indeed, it is not necessary to change the size of the display so often. A floating status bar would be a really good solution.

**Comment by RannyArcher** [comment]
@The-Compiler 

i just go around the source code, i notice that the bar shown by `message.info` and other methodes in the same module (`message`)... the string passed to those functions is shown on floating bars and doesn't cause shifting layout to the webpage, and that what most of users want. the only difference is that this bar is temporary disappearing.

i wish if there is a way to make that floating option can be turned `true` or `false` according to user's config. cause imposing it and let it be the only option available will cause a part of that site to be hidden behind the floating bar.

or to be even specific this behavior is only wanted when:

`tab.show = switching`
and
`statusbar.show = in-mode`

i'm sure no user want the shifting layout to happen when the tabs/status bar  are going to hide. without adding the floating option and add more burden to the project and devloper's headache.

i really tried to fix that and collaborate to this great project. but i have difficulties cause i'm not that good python advanced devloper.

wish you prioritize this issue to be fixed, and i'm sure it will not take long time by your hands Mr.@The-Compiler . and i'm sure more people will have good user experience than before.

wish you work on that in your empty time, thanks for this perfect qutebrowser which makes us save lot of time.

**Comment by linkert** [comment]
Any new possibilities opened up regarding this issue now that were on Qt6? (Or maybe this [issue](https://github.com/qutebrowser/qutebrowser/issues/5654)?)

Just about three years since I mentioned and illustrated this [issue](https://github.com/qutebrowser/qutebrowser/issues/5748). Been mostly rocking Firefox during this time with vertical tab bar that expands and contracts and does not resize the website being viewed – but Firefox lacks in so many other ways and I'm in the browser market again during my new NixOS adventures, once again considering my old loved qutebrowser to be configured via my home manager on NixOS.

It's so dizzying when you are trying to check the content between two tabs and everything jumps from side to side (and I guess up and down if you have hidden tab bar positioned on the top or bottom). I remember that day three years ago feeling my eyes straining and thus prompting me to post the issue. To me, the issue contrasts with the priority level being set to "low" and I imagine many others it's rather critical, a deal breaker.

Came to ask about Qt6, the initial question of my post, and sort of bump the issue but went on a tangent. Love the work, wish I could help with something more concrete (not a dev) and I hope this message does not take away from my appreciation of your work @The-Compiler .

**Comment by The-Compiler** [comment]
Sorry for the late answer @linkert, this got lost in my inbox for a while. I don't think Qt 6 changes anything here. Happy to bump up the priority a bit since this indeed seems to be coming up every once in a while. Thanks for the nice words, much appreciated!

**Comment by zbyyt** [comment]
Perhaps a related subject to this issue.

As a quickfix for the page jolting problem in relation to a hidden statusbar, I have set my status widgets to an empty dictionary so an empty black bar keeps the page in place whilst I exit normal mode.

The main issue I have for the momentary jolt was that sometimes when entering hint mode, the appearance of the bar and the subsequent jolt in page would occur as the hints were being generated, which meant the hints appeared unaligned with the page elements.

I also have tabbar.show set to switching, but now the tabs appear above the empty statusbar placeholder. Whilst before the tabbar would elegantly appear in the same place as the other auto hiding bars, because the bars would always be hidden during tab switching.

Having the option to configure floating bars would obviously be ideal, but also if this is to be developed it may be desirable to decide if bars float over each other, like the message bar does over the statusbar, or whether they float in a non-overlapping order.

For me, the main priority is the jolting that relates to entering hint mode, as the confusion in hint placement significantly slows my browsing speed as I navigate. For example when navigating cumbersome multipage login processes.

The location of the tab bar on switching is secondary. Also, it obviously still creates a page jolt on some pages but I rarely notice it as a functional problem in comparison to the auto hiding statusbar.

**Comment by The-Compiler** [comment]
As pointed out by @emanuele6 in IRC, this is also relevant when hovering links with chinese/arabic/non-latin characters. Currently this causes the statusbar to resize, which also causes the page to shift.

**Comment by emanuele6** [comment]
https://github.com/user-attachments/assets/199f88f7-b4ea-4031-acfb-e2362b12edaf

**Comment by log-lot** [comment]
I hope this gets considered to be worked on soon, as in my experience some pages partially / temporarily break when a large page shift happens (like the left/right aligned tabbar hiding on the switching mode) and theming possibilities with transparency opens up when the tabbar is over the page contents rather than shifting them, though it should definitely be togglable, as some people could prefer always having the tabbar open and floating, or having the tabbar open on switch and move the page contents
